internal/controllers/bundledeployment/bundledeployment.go

```diff
@@ -198,53 +249,79 @@ func (c *controller) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 // Today we always return ctrl.Result{} and an error.
 // But in the future we might update this function
 // to return different results (e.g. requeue).
-func (c *controller) reconcile(ctx context.Context, bd *rukpakv1alpha1.BundleDeployment) (ctrl.Result, error) {
+func (c *controller) reconcile(ctx context.Context, bd *rukpakv1alpha2.BundleDeployment) (ctrl.Result, error) {
 	bd.Status.ObservedGeneration = bd.Generation
 
-	bundle, allBundles, err := util.ReconcileDesiredBundle(ctx, c.cl, bd)
+	// handle finalizers
+	finalizerBundleDelpoyment := bd.DeepCopy()
```

Comment:
Do we need another deep copy of the bundle deployment here? We deep copy in `Reconcile` already.

Reply:
This is required because we are finalizing the deepCopy, and if there is any change, updating the finalizers: https://github.com/operator-framework/rukpak/blob/4f54625e9eec52b211fdaf1c41c00b86715b8022/internal/controllers/bundledeployment/bundledeployment.go#L271-L278

Reply:
In both of those cases, we end up returning the finalized BD anyway though. It seems like the only thing this approach "buys" us (if it's even a benefit) is that we only use the updated finalizers and status. If the finalizer implementation changes anything else, we would ignore it with this approach.

We handle this a bit differently in operator-controller. Perhaps we copy what it does? (i.e. let code change whatever it wants, and then assert/panic if unexpected fields changed): https://github.com/operator-framework/operator-controller/blob/main/internal/controllers/clusterextension_controller.go#L87-L89